src/export/export_format_spaten.cpp

```diff
@@ -0,0 +1,249 @@
+        proto_feat.add_message(spaten_pbf::Feature::optional_Tag_tags, tagbuf);
+        tagbuf.clear();
+    }
+
+    if (!options().id.empty()) {
+        ptag.add_string(spaten_pbf::Tag::optional_string_key, options().id);
+        ptag.add_string(spaten_pbf::Tag::optional_string_value, uint64_buf(object.type() == osmium::item_type::area ? osmium::area_id_to_object_id(object.id()) : object.id()));
+        ptag.add_enum(spaten_pbf::Tag::optional_ValueType_type, spaten_pbf::TagValueType::uint64);
+        proto_feat.add_message(spaten_pbf::Feature::optional_Tag_tags, tagbuf);
+        tagbuf.clear();
+    }
+
+    if (!options().version.empty()) {
+        ptag.add_string(spaten_pbf::Tag::optional_string_key, options().version);
+        ptag.add_string(spaten_pbf::Tag::optional_string_value, uint64_buf(object.version()));
+        ptag.add_enum(spaten_pbf::Tag::optional_ValueType_type, spaten_pbf::TagValueType::uint64);
+        proto_feat.add_message(spaten_pbf::Feature::optional_Tag_tags, tagbuf);
+        tagbuf.clear();
+    }
+
+    if (!options().changeset.empty()) {
+        ptag.add_string(spaten_pbf::Tag::optional_string_key, options().changeset);
+        ptag.add_string(spaten_pbf::Tag::optional_string_value, uint64_buf(object.changeset()));
+        ptag.add_enum(spaten_pbf::Tag::optional_ValueType_type, spaten_pbf::TagValueType::uint64);
+        proto_feat.add_message(spaten_pbf::Feature::optional_Tag_tags, tagbuf);
+        tagbuf.clear();
+    }
+
+    if (!options().uid.empty()) {
+        ptag.add_string(spaten_pbf::Tag::optional_string_key, options().uid);
+        ptag.add_string(spaten_pbf::Tag::optional_string_value, uint64_buf(object.uid()));
+        ptag.add_enum(spaten_pbf::Tag::optional_ValueType_type, spaten_pbf::TagValueType::uint64);
+        proto_feat.add_message(spaten_pbf::Feature::optional_Tag_tags, tagbuf);
+        tagbuf.clear();
+    }
+
+    if (!options().user.empty()) {
+        protozero::pbf_builder<spaten_pbf::Tag> ptag{tagbuf};
+        ptag.add_string(spaten_pbf::Tag::optional_string_key, options().user);
+        ptag.add_string(spaten_pbf::Tag::optional_string_value, object.user());
+        proto_feat.add_message(spaten_pbf::Feature::optional_Tag_tags, tagbuf);
+        tagbuf.clear();
+    }
+
+    if (!options().timestamp.empty()) {
+        ptag.add_string(spaten_pbf::Tag::optional_string_key, options().timestamp);
+        ptag.add_string(spaten_pbf::Tag::optional_string_value, uint64_buf(object.timestamp().seconds_since_epoch()));
+        ptag.add_enum(spaten_pbf::Tag::optional_ValueType_type, spaten_pbf::TagValueType::uint64);
+        proto_feat.add_message(spaten_pbf::Feature::optional_Tag_tags, tagbuf);
+        tagbuf.clear();
+    }
+
+    if (!options().way_nodes.empty() && object.type() == osmium::item_type::way) {
+        for (const auto& nr : static_cast<const osmium::Way&>(object).nodes()) {
```

Comment:
This creates several "tags" with the same key, each with one node id as value. Shouldn't this be one "tag" with a list of nodes? (Not sure what type if would have, we'd need some kind of array type, maybe just "stringify" this? Or use a 64bit length N plus N x 64bit id?) The current code violates the assumption that there is only one tag with a given key.

Reply:
For now, I switched to a space separated string. Probably some array type would be useful at some point, but so far I don't have a solid idea for that.